### PR TITLE
provider/aws: Randomize test names

### PR DESCRIPTION
**Pull request by mbfrahry**
This should finish up the rest of the tests that failed because of duplicate names

```
make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSEMRCluster'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/03/28 15:27:12 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSEMRCluster -timeout 120m
=== RUN   TestAccAWSEMRCluster_basic
--- PASS: TestAccAWSEMRCluster_basic (661.11s)
=== RUN   TestAccAWSEMRCluster_terminationProtected
--- PASS: TestAccAWSEMRCluster_terminationProtected (535.72s)
=== RUN   TestAccAWSEMRCluster_visibleToAllUsers
--- PASS: TestAccAWSEMRCluster_visibleToAllUsers (529.23s)
=== RUN   TestAccAWSEMRCluster_tags
--- PASS: TestAccAWSEMRCluster_tags (562.33s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/aws	2288.426s
```

```
make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSPolicyAttachment_paginatedEntities'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/03/29 14:18:39 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSPolicyAttachment_paginatedEntities -timeout 120m
=== RUN   TestAccAWSPolicyAttachment_paginatedEntities
--- PASS: TestAccAWSPolicyAttachment_paginatedEntities (112.23s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/aws	112.271s

```

```
make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSVpnConnection'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/03/29 14:41:49 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSVpnConnection -timeout 120m
=== RUN   TestAccAWSVpnConnection_importBasic
--- PASS: TestAccAWSVpnConnection_importBasic (227.27s)
=== RUN   TestAccAWSVpnConnection_basic
--- PASS: TestAccAWSVpnConnection_basic (416.36s)
=== RUN   TestAccAWSVpnConnection_withoutStaticRoutes
--- PASS: TestAccAWSVpnConnection_withoutStaticRoutes (242.45s)
=== RUN   TestAccAWSVpnConnectionRoute_basic
--- PASS: TestAccAWSVpnConnectionRoute_basic (230.49s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/aws	1116.608s
```

### CODE REVIEW

File: builtin/providers/aws/resource_aws_iam_policy_attachment_test.go [hunk 1]
```diff
@@ -303,9 +307,13 @@ resource "aws_iam_policy_attachment" "test-attach" {
...
 
 const testAccAWSPolicyPaginatedAttachConfig = `
+resource "random_id" "user_id" {
+  byte_length = 10
+}
+
 resource "aws_iam_user" "user" {
     count = 101
-    name = "${format("paged-test-user-%d", count.index + 1)}"
+    name = "${format("paged-test-user-${random_id.user_id.hex}-%d", count.index + 1)}"
```
Comment (mbfrahry) [inline builtin/providers/aws/resource_aws_iam_policy_attachment_test.go]: I was trying to figure this out and what you said makes total sense. I'll make the changes

File: builtin/providers/aws/resource_aws_vpn_connection_test.go [hunk 2]
```diff
@@ -161,47 +164,55 @@ func TestAWSVpnConnection_xmlconfig(t *testing.T) {
...
-  bgp_asn = 65000
-  ip_address = "178.0.0.1"
-  type = "ipsec.1"
-}
+	resource "aws_customer_gateway" "customer_gateway" {
+	  bgp_asn = 65000
+	  ip_address = "178.0.0.1"
+	  type = "ipsec.1"
+		tags {
+	    Name = "main-customer-gateway-%d"
```
Comment (mbfrahry) [inline builtin/providers/aws/resource_aws_vpn_connection_test.go]: I'll get a randomly generated ASN in here! Thanks for finding this.

**Comment by mbfrahry** [comment]
@radeksimko PTAL

**Comment by radeksimko** [comment]
Thanks @mbfrahry this looks good. I also found another thing to fix, so I just committed to your branch, I hope you don't mind.

I ran all relevant acceptance tests again in parallel and they all pass so this now LGTM.

**Comment by mbfrahry** [comment]
Thanks @radeksimko!

**Pull request merged**